File: app/src/main/java/me/toptas/fancyshowcasesample/AnimatedActivity.java
package me.toptas.fancyshowcasesample;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.Button;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import me.toptas.fancyshowcase.FancyShowCaseQueue;
import me.toptas.fancyshowcase.FancyShowCaseView;
import me.toptas.fancyshowcase.OnViewInflateListener;


public class AnimatedActivity extends BaseActivity {

    @BindView(R.id.btn_focus)
    Button mButton;

    @BindView(R.id.btn_focus2)
    Button mButton2;


    private FancyShowCaseQueue mQueue;
    private FancyShowCaseView mFancyView, mFancyView2;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_animated);
        ButterKnife.bind(this);

        mFancyView = new FancyShowCaseView.Builder(this)
                .focusOn(mButton)
                .customView(R.layout.layout_animated_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(@NonNull View view) {
                        setAnimatedContent(view, mFancyView);
                    }
                })
                .build();

        mFancyView2 = new FancyShowCaseView.Builder(this)
                .focusOn(mButton2)
                .customView(R.layout.layout_animated_view, new OnViewInflateListener() {
                    @Override
                    public void onViewInflated(@NonNull View view) {
                        setAnimatedContent(view, mFancyView2);
                    }
                })
                .build();


    }

    private void setAnimatedContent(final View view, final FancyShowCaseView fancyShowCaseView) {
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                TextView tvMain = (TextView) view.findViewById(R.id.tvMain);
                final TextView tvSub = (TextView) view.findViewById(R.id.tvSub);
                TextView tvNext = (TextView) view.findViewById(R.id.btn_next);
                TextView tvDismiss = (TextView) view.findViewById(R.id.btn_dismiss);

                if (fancyShowCaseView.equals(mFancyView2)) {
                    tvMain.setText("My Fancy Title 2");
                    tvSub.setText("My fancy description can be a longer text.2");
                    tvNext.setText("Close");
                }

                tvNext.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        fancyShowCaseView.hide();
                    }
                });

                tvDismiss.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        mQueue.cancel(true);
                    }
                });

                final Animation mainAnimation = AnimationUtils.loadAnimation(AnimatedActivity.this, R.anim.slide_in_left);
                mainAnimation.setFillAfter(true);

                final Animation subAnimation = AnimationUtils.loadAnimation(AnimatedActivity.this, R.anim.slide_in_left);
                subAnimation.setFillAfter(true);
                tvMain.startAnimation(mainAnimation);
                new Handler().postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        tvSub.startAnimation(subAnimation);

                    }
                }, 80);
            }
        }, 200);
    }

    @OnClick(R.id.btn_focus)
    public void focus(View view) {
        mQueue = new FancyShowCaseQueue();
        mQueue.add(mFancyView);
        mQueue.add(mFancyView2);
        mQueue.show();
    }
}

File: app/src/main/java/me/toptas/fancyshowcasesample/AnimatedActivity.kt
/*
 * Copyright (c) 2018. Faruk Toptaş
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.toptas.fancyshowcasesample

import android.os.Bundle
import android.os.Handler
import android.view.View
import android.view.animation.AnimationUtils
import android.widget.TextView

import me.toptas.fancyshowcase.FancyShowCaseQueue
import me.toptas.fancyshowcase.FancyShowCaseView
import me.toptas.fancyshowcase.listener.OnViewInflateListener
import me.toptas.fancyshowcasesample.databinding.ActivityAnimatedBinding


class AnimatedActivity : BaseActivity() {

    private lateinit var binding: ActivityAnimatedBinding
    private lateinit var queue: FancyShowCaseQueue
    private lateinit var fancyView: FancyShowCaseView
    private lateinit var fancyView2: FancyShowCaseView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAnimatedBinding.inflate(layoutInflater)
        setContentView(binding.root)

        fancyView = FancyShowCaseView.Builder(this)
                .focusOn(binding.btnFocus)
                .customView(R.layout.layout_animated_view, object : OnViewInflateListener {
                    override fun onViewInflated(view: View) {
                        setAnimatedContent(view, fancyView)
                    }
                })
                .build()

        fancyView2 = FancyShowCaseView.Builder(this)
                .focusOn(binding.btnFocus2)
                .customView(R.layout.layout_animated_view, object : OnViewInflateListener {
                    override fun onViewInflated(view: View) {
                        setAnimatedContent(view, fancyView2)
                    }
                })
                .build()

        binding.btnFocus.setOnClickListener {
            queue = FancyShowCaseQueue().apply {
                add(fancyView)
                add(fancyView2)
                show()
            }
        }


    }

    private fun setAnimatedContent(view: View, fancyShowCaseView: FancyShowCaseView) {
        Handler().postDelayed({
            val tvMain = view.findViewById<View>(R.id.tvMain) as TextView
            val tvSub = view.findViewById<View>(R.id.tvSub) as TextView
            val tvNext = view.findViewById<View>(R.id.btn_next) as TextView
            val tvDismiss = view.findViewById<View>(R.id.btn_dismiss) as TextView

            if (fancyShowCaseView == fancyView2) {
                tvMain.text = "My Fancy Title 2"
                tvSub.text = "My fancy description can be a longer text.2"
                tvNext.text = "Close"
            }

            tvNext.setOnClickListener { fancyShowCaseView.hide() }

            tvDismiss.setOnClickListener { queue.cancel(true) }

            val mainAnimation = AnimationUtils.loadAnimation(this@AnimatedActivity, R.anim.slide_in_left)
            mainAnimation.fillAfter = true

            val subAnimation = AnimationUtils.loadAnimation(this@AnimatedActivity, R.anim.slide_in_left)
            subAnimation.fillAfter = true
            tvMain.startAnimation(mainAnimation)
            Handler().postDelayed({ tvSub.startAnimation(subAnimation) }, 80)
        }, 200)
    }
}
